secant/Features/BackupFlow/RecoveryPhraseValidation.swift

```diff
@@ -0,0 +1,229 @@
+//
+//  RecoveryPhraseValidation.swift
+//  secant-testnet
+//
+//  Created by Francisco Gindre on 10/29/21.
+//
+
+import Foundation
+import ComposableArchitecture
+
+typealias RecoveryPhraseValidationStore = Store<RecoveryPhraseValidationState, RecoveryPhraseValidationAction>
+typealias RecoveryPhraseValidationViewStore = ViewStore<RecoveryPhraseValidationState, RecoveryPhraseValidationAction>
+
+struct RecoveryPhraseEnvironment {
+    var mainQueue: AnySchedulerOf<DispatchQueue>
+    var newPhrase: () -> Effect<RecoveryPhrase, AppError>
+}
+
+/// Represents the completion of a group of recovery words by de addition of one word into the given group
+struct RecoveryPhraseStepCompletion: Equatable {
+    var groupIndex: Int
+    var word: String
+}
+
+struct RecoveryPhraseValidationState: Equatable {
+    enum Step: Equatable {
```

Comment:
I think this is less of a step than it is a status or a phase.  Also, what is the semantic difference between complete/invalid/valid?  If the status is complete, aren't you intrinsically either valid or invalid?  Or rather, if you are valid or invalid, aren't you automatically complete?

Reply:
Thanks for bringing this up. I ran out of names! Suggestions are very welcome :)

> Also, what is the semantic difference between complete/invalid/valid? If the status is complete, aren't you intrinsically either valid or invalid? Or rather, if you are valid or invalid, aren't you automatically complete?
This comes from a train of thought of driving the state for the screens. It's possible that either the complete or the valid/invalid states are not needed.

I initially thought of `complete` as the stage where you have completed and now the transition to failure screen or success screen has to be carried on. I don't know how you can actually do that elegantly thorough the Reducer so tha when completions equal the amount of groups you can drive the flow forward and trigger some action.  This comes more from my lack of understanding of TCA than something else.

Reply:
What did we think about `Status` here?